recipes-devtools/rust/rust.inc

```diff
@@ -315,6 +366,10 @@ def rust_gen_mk_cfg(d, thing):
         '-e', '/^CFG_GCCISH_CXXFLAGS/ s;$; {};'.format(cxxflags_for(d, thing)),
         '-e', '/^CFG_GCCISH_LINK_FLAGS/ s;$; {};'.format(ldflags),
 
+        # The mipsel target specify flags for the mips32. Replace them with
+        # mips32r2 if that's what we use
+        '-e', 's/mips32/mips32r2/g' if is_mips32r2 else '',
```

Comment:
I'm not entirely comfortable with this type of substitution. All the existing ones explicitly seek out variables to set, this should too.

On ldflags: are the ones that yocto provides in `LDFLAGS`, `HOST_CC_ARCH` and `TOOLCHAIN_OPTIONS` not already sufficient? Or is the goal here to remove some bad ones?